### ISSUE TITLE
Help with make install instructions for the compilation

### ISSUE THREAD

**Issue by marionpavaux** [ecr05/MSM_HOCR_macOSX#20]
Hi, I got some difficulties with the installation of the MSM. 
I precisely followed the compilation instructions and got to the point where I had to do the make install for the DiscreteOpt, FastPDlib, etc.. 

The thing is, the compiler could not find the "includes" like includes related to the file newmesh, etc. It cannot find it because it is not the same folder. 
Another issue is with the FastPDlib, the error was it could not find the target. 

I am just wondering if you have any idea of what could got wrong during the installation , or more detailed instructions we have to follow for "make install". 

Thanks !

**Comment by jewellewej** [comment]
Hi Emma, I got the same issues when I tried to compile MSM, can you provide any further comments? After I changed the path of newmesh folder, other similar problems occurred.  I've tried the compiled version you released, but it didn't work (my OS is Ubuntu 18.04).

**Comment by ecr05** [comment]
newmesh should be in $FSLDIR/include/newmesh/ . I built my FSL from source. If it's not there you will need to post to the FSL list to ask them where to find it. Have you added the path to $FSLDIR to your path as per: https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FslInstallation/SourceCode#Compiling_Source_Distribution which says to add these two lines to your environment script

export FSLDIR=`pwd`/fsl
. ${FSLDIR}/etc/fslconf/fsl.sh

If the ubuntu 16 version doesn't work then I'll probably have to rebuild msm for ubuntu 18 myself soon and add it to the release tab. Probably with the next 2 weeks.

No the FSL version isn't enough.

**Comment by marionpavaux** [comment]
Hi, I think finally that the problem comes from the fact the fsl is not building properly. It seems not all projects are built. 
So finally I tried to download the binaries directly with the msm_ubuntu_v3. 
I then made chmod +x msm_ubuntu_v3 in the terminal to gives the authorization to execute. 
Then when I try ./msm_ubuntu_v3 --help, I get the error: 

./msm_ubuntu_v3 : error while loading shared libraries: libopenblas.so.0: cannot open shared object file: no such file or directory. 

So I really don't know what to do. Either I try to install the msm but the fsl is not built correctly, or I download the binaries but it is not working as well. Have you any idea ? 

Also can I use the dHCP_template_alignment with the binaries If then I succeed to execute?

**Comment by ecr05** [comment]
I need to rebuild msm for Ubuntu 18 for my systems.  I'll get to it by beginning of next week and add it to the release

**Comment by marionpavaux** [comment]
I have Ubuntu 20.04 in case it needs more. Thanks for your help.

**Comment by ecr05** [comment]
I don't have Ubuntu 20 to build it there I'm afraid.  

Don't you have IT support that can help you build FSL and then MSM from source? Or downgrade your Ubuntu?

You can also check with the FSLlist for compile support. I cannot provide you with any more guidance than is provided on their website.

**Comment by marionpavaux** [comment]
Yes it seems I will have to understand why fsl is not building correctly with their website. 
Thanks

**Comment by ecr05** [comment]
HI @marionpavaux @jewellewej. It turns out that building for ubuntu18 is really not working at the moment. As a workaround you can download the msm_centos_v3 binary but you will either have to put it in your ${FSLDIR}/bin folder and/or install the blas library using:

sudo apt-get install libopenblas-dev

To get it to then run.

**Comment by marionpavaux** [comment]
Hi, thanks for your comment, we finally succeeded in making the binaries msm_ubuntu_v3 work in a ubuntu 16 container using singluarity. It turns out finally that it is also working for ubuntu 20. The library was in fact not installed so it was not working at the beginning. 
The only thing now is when I try to run it with the Neuroimage aMSM configuration file and not the Neuroimage normal MSM configuration file: 

I get the error 

cost.size() 3 DISCRETE,DISCRETE,DISCRETE
 iters 40,40,40
lambda 0.025,0.025,0.025
 sigmain 6,4,2
 sigmaref 6,4,2
 datagrid 4,5,6
 cp gridres 2,3,4
 sp grid 4,5,6
 alpha_knn 5,5,5
 in parse options  _simval 2,2,2 interations 40,40,40 input sigma 6,4,2 ref sigma 6,4,2 lambda 0.025,0.025,0.025  _genesis 4,5,6 _gridres 2,3,4 _sampres 4,5,6 opt DISCRETE,DISCRETE,DISCRETE _scale 0 mesh interpolator BARY discrete implementation HOCR regoption 5 potts 0
 Initialising level 1
terminate called after throwing an instance of 'MESHREG::MeshReg_error'
  what():  ��Q
Aborted

**Comment by ecr05** [comment]
I would need to see your call to msm

**Comment by ecr05** [comment]
You are missing the anatomical surfaces which are required for aMSM see https://emmarobinson01.com/2020/04/05/advanced-features-of-msm-a-guide/
